src/api/ingest/serializers.py

```diff
@@ -0,0 +1,511 @@
+        elif isinstance(data, Sequence) and isinstance(data, str) is False:
+            self.original_data = [d for d in data]
+        else:
+            self.original_data = None
+
+        if project_choices:
+            self.project_choices = project_choices
+
+        super().__init__(instance=None, data=data, **kwargs)
+
+    @classmethod
+    def many_init(cls, *args, **kwargs):
+        if "data" in kwargs and isinstance(kwargs["data"], (dict, OrderedDict)):
+            kwargs["data"] = [kwargs["data"]]
+        return super(CollectRecordCSVSerializer, cls).many_init(*args, **kwargs)
+
+    def get_initial(self):
+        if not isinstance(self._original_data, Mapping):
+            return OrderedDict()
+
+        return OrderedDict(
+            [
+                (field_name, field.get_value(self._original_data))
+                for field_name, field in self.fields.items()
+                if (field.get_value(self._original_data) == empty)
+                and not field.read_only
+            ]
+        )
+
+    def validate(self, data):
+        # Validate common Transect level fields
+        return data
+
+    def create_path(self, field_path, node, val):
+        path = field_path.pop(0)
+        if path not in node:
+            if len(field_path) >= 1:
+                node[path] = dict()
+                self.create_path(field_path, node[path], val)
+            else:
+                node[path] = val
+        else:
+            if len(field_path) >= 1:
+
+                self.create_path(field_path, node[path], val)
+            else:
+                node[path] = val
+        return node
+
+    def create(self, validated_data):
+        output = validated_data.copy()
+
+        for name, field in self.fields.items():
+            field_path = field.field_name.split("__")
+            val = validated_data.get(name)
+            output = self.create_path(field_path, output, val)
+
+        # Need to serialize observers after validation to avoid
+        # unique id errors
+        project_profiles = self.project_choices.get("project_profiles")
+        ob_names = output["data"]["observers"]
+        output["data"]["observers"] = [
+            project_profiles.get(ob_name.lower()) for ob_name in ob_names
+        ]
+
+        return output
+
+    def format_error(self, record_errors):
+        fmt_errs = dict()
+        field_map = dict(zip(self.header_map.values(), self.header_map.keys()))
+        for k, v in record_errors.items():
+            display = field_map.get(k) or k
+            fmt_errs[display] = {"description": v}
+        return fmt_errs
+
+    @property
+    def formatted_errors(self):
+        errors = self.errors
+        return self.format_error(errors)
+
+
+class BenthicPITCSVSerializer(CollectRecordCSVSerializer):
+    protocol = "benthicpit"
+    observations_field = "data__obs_benthic_pits"
+    ordering_field = "data__obs_benthic_pits__interval"
+    header_map = CollectRecordCSVSerializer.header_map
```

Comment:
Can't this be `self.header_map`?

Reply:
self isn't defined yet, header_map is first inheriting `header_map` from `CollectRecordCSVSerializer` and then extending on the next line